src/core/types.ts

```diff
@@ -6,11 +6,13 @@
 export interface Pkg {
   name: string;
   version?: string;
+  url?: string;
```

Comment:
it should probably not go here? 
as `Pkg` is like the identifier of a package, and `PkgInfo` is meant to contain extra info

Reply:
Sooooooooooooo....... yeh. Except in this case `url` is supposed to be part of the identity. But it's becoming clear to me that this is more involved than we thought (if `url` is part of the identity then it needs to be part of package id etc).